testcafe/tests/pages/drive-model.js

```diff
@@ -29,53 +32,129 @@ export default class Page {
     this.driveEmpty = Selector('[class*="c-empty"]')
       .parent(0)
       .withAttribute('data-test-id', 'fil-content-body')
-
     //oops is a class for error only
     this.errorOops = Selector('[class*="oops"]')
-
     this.errorEmpty = Selector('[class*="c-empty"]')
       .child('h2')
       .withText('Switch online!') // !FIXME: do not use text
+    //loading
+    this.contentPlaceHolder = Selector(
+      '[class*="fil-content-file-placeholder"]'
+    )
+    this.alertWrapper = Selector('[class*="c-alert-wrapper"]')
 
-    //Toolbar
-    this.toolbarFile = getElementWithTestId('fil-toolbar-files')
-    this.btnMoreMenu = this.toolbarFile
-      .find('button')
-      .withAttribute('title', 'more')
+    //Toolbar - Action Menu
+    this.toolbarFiles = getElementWithTestId('fil-toolbar-files')
+    this.btnMoreMenu = this.toolbarFiles.find('[class*="dri-btn--more"]')
+    this.cozyMenuInner = getElementWithTestId('coz-menu-inner')
     this.btnAddFolder = getElementWithTestId('add-folder-link').parent(
-      'div.[class*="coz-menu-item"]'
+      '[class*="coz-menu-item"]'
+    )()
+    this.btnRemoveFolder = getElementWithTestId('fil-action-delete')
+
+    //Files list
+    this.content_rows = Selector(
+      `[class*="fil-content-row"]:not([class*="fil-content-row-head"])`
+    )
+    this.foldersNamesInputs = getElementWithTestId('name-input')
+    this.foldersNames = getElementWithTestId('fil-file-filename-and-ext')
+    this.checboxFolderByRowIndex = value => {
+      return this.content_rows
+        .nth(value)
+        .child('[class*="fil-content-cell"]')
+        .child('[data-input="checkbox"]')
+    }
+
+    // Upload
+    this.btnUpload = getElementWithTestId('uploadButton')
+    this.divUpload = getElementWithTestId('upload-queue')
+    this.divUploadSuccess = getElementWithTestId('upload-queue-success')
+
+    // Sharing
+    this.btnShare = this.toolbarFiles
+      .child('button')
+      .withAttribute('data-test-id', 'share-button')
+    this.divShareByLink = getElementWithTestId('share-by-link')
+    this.toggleShareLink = this.divShareByLink.child('[class*="toggle"]')
+    this.spanLinkCreating = Selector('[class*="share-bylink-header-creating"]')
+    this.copyBtnShareByLink = Selector('button').withAttribute('data-test-url')
+    this.btnShareByMe = this.toolbarFiles
+      .child('button')
+      .withAttribute('data-test-id', 'share-by-me-button')
+
+    //Top Option bar & Confirmation Modal
+    this.cozySelectionbar = Selector('[class*="coz-selectionbar"]')
+    this.cozySelectionbarBtnDelete = this.cozySelectionbar
+      .find('button')
+      .withText('REMOVE') //!FIX ME : do not use text! Do not use .nth(x) because the buttons count changes if one or several files/folders are selected
+    this.modalDelete = Selector('[class*="c-modal"]').find('div')
+    this.modalDeleteBtnDelete = this.modalDelete.find('button').nth(2) //REMOVE
+  }
+
+  async waitForLoading() {
```

Comment:
Even if the method doesn't take any argument, you can describe what the method does ;)